modules/web/src/main/scala/Application.scala

```diff
@@ -37,6 +41,21 @@ object Application {
         val limit   = n.getOrElse(100)
         gs.queryProgramsByName(pattern, limit).flatMap(ps => Ok(ps.map(p => (p._1, p._2)).asJson ))
 
+      // Import an observation from the legacy ODB.
+      case GET -> Root / "api" / "import" / "obs" / o :? Host(h) as gs =>
+        withObsId(o) { oid =>
+          gs.ocs2.importObservation(h, oid).flatMap {
+            _.fold(msg => InternalServerError(msg), _ => NoContent())
+          }
+        }
+
+      // Select an observation by id.
+//      case GET -> Root / "api" / "fetch" / "obs" / o as gs =>
+//        withObsId(o) { oid =>
+//          gs.queryObservationById(oid).flatMap { obs =>
+//            Ok(obs.asJson)
```

Comment:
Are we doing these things in JSON?  What would I need to do to get the codec for an `Observation`?  I assume it is derived because there is no explicit codec in the `gem.json` package but it isn't being picked up.

Reply:
isn't the codec defined in #535 ?

Reply:
D'oh, you're right.  I keep forgetting that is still pending.